src/App.tsx

```diff
@@ -15,48 +16,119 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+interface FilterParams {
+  sortField: string;
+  isReversed: boolean;
+}
+
+enum SortField {
+  Alph = 'Sort alphabetically',
+  Lngth = 'Sort by length',
+  None = 'none',
+}
+
+function getPreparedList(
+  goods: string[],
+  { sortField, isReversed }: FilterParams,
+) {
+  let prepareGoods = [...goods];
+
+  if (sortField !== SortField.None) {
+    prepareGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortField.Alph:
+          return good1.localeCompare(good2);
+
+        case SortField.Lngth:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    prepareGoods = prepareGoods.reverse();
+  }
+
+  return prepareGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortField.None);
+  const [isReversed, setIsReversed] = useState(false);
+  const listOfGoods
+    = getPreparedList(goodsFromServer, { sortField, isReversed });
+
+  const handleSortByAlph = () => {
+    setSortField(SortField.Alph);
+  };
+
+  const handleSortByLngth = () => {
+    setSortField(SortField.Lngth);
+  };
```

Comment:
```suggestion
  const makeHandleSortBy = (field: SortField) => () => {
    setSortField(field);
  };
  
```
better to create one handler for two sort field.